src/seedu/addressbook/parser/Parser.java

```diff
@@ -50,36 +50,37 @@ public Parser() {}
      * @param userInput full user input string
      * @return the command based on the user input
      */
-    public Command parseCommand(String userInput) {
+    public static Command parseCommand(String userInput) {
         final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
         if (!matcher.matches()) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
         }
-
+        Parser parse = new Parser();
```

Comment:
This works too but there is no point creating an object here. The better alternative is to convert all methods to static.